### PR TITLE
Subscriptions Engine

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

Changes `SubscriptionsMechanism` to `SubscriptionsEngine`. Moves out of the `plugins` directory to fully move away from this concept.

## Complexity

Complexity: Medium

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-3688.surge.sh

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟥 query.QueryWithNestedIn | 14848 | 13333 | 65 | 84 | 1698 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 43 | 51 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 86 | 124 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 71 | 81 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 4538 | 4330 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3862 | 3591 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 55 | 78 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 54 | 75 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070331 | 6259 | 5748 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 192 | 279 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 149 | 93 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 254 | 249 | 3003 |
|  connections.Connection | 14082 | 14082 | 101 | 95 | 2174 |
|  connections.NestedConnection | 45472 | 45472 | 179 | 164 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 94 | 97 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 121 | 156 | 2174 |
|  create.SimpleMutation | 7 | 7 | 74 | 81 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 61 | 43 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 410 | 511 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 284 | 284 | 1179 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 57 | 142 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 117 | 148 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 57 | 55 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 94 | 79 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 34 | 28 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 64 | 65 | 2174 |
|  query.QueryWhere | 9711 | 9695 | 48 | 53 | 2167 |
|  query.SimpleQueryWithNestedWhere | 9889 | 9873 | 75 | 79 | 2167 |
|  query.Nested | 10096041 | 10096041 | 11748 | 11427 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 11304 | 11611 | 2004000 |
|  query.OrFilterOnRelationships | 42953 | 42692 | 247 | 316 | 1735 |
|  query.OrFilterOnRelationshipsAndNested | 35976 | 36593 | 281 | 374 | 1735 |
| 🟥 query.QueryWithNestedIn | 14848 | 13333 | 65 | 84 | 1698 |
|  query.NestedConnectionWhere | 9834 | 9834 | 74 | 93 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10020 | 9896 | 98 | 120 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19254 | 19116 | 165 | 317 | 1636 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 244 | 349 | 1134 |
|  query.Fulltext | 80 | 80 | 70 | 41 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 63 | 60 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 179 | 92 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14534 | 14491 | 140 | 137 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 71 | 75 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 121 | 137 | 4516 |
|  sorting.SortWithTopLevelCypher | 16162 | 16162 | 62 | 67 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 81 | 82 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 69 | 81 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 329 | 360 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 322 | 365 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 128 | 185 | 2002 |

</details>
        
Old Schema Generation: 58.154s
Schema Generation: 59.461s
Old Subgraph Schema Generation: 1:11.895 (m:ss.mmm)
Subgraph Schema Generation: 1:10.386 (m:ss.mmm)

File: examples/neo-place/create-plugin.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const { Neo4jGraphQLSubscriptionsAMQPPlugin } = require("@neo4j/graphql-plugin-subscriptions-amqp");
+const { Neo4jGraphQLSubscriptionsEngineAMQP } = require("@neo4j/graphql-subscriptions-engine-amqp");
```
Comment (angrykoala) [inline examples/neo-place/create-plugin.js:1]: Remember to deploy this to the live demo at https://team-graphql.uc.r.appspot.com/

File: examples/subscriptions/apollo_rabbitmq/server.js [hunk 2]
```diff
@@ -27,15 +27,15 @@ const { useServer } = require("graphql-ws/lib/use/ws");
...
+const { Neo4jGraphQLSubscriptionsEngineAMQP } = require("@neo4j/graphql-subscriptions-engine-amqp");
 
 const NEO4J_URL = "bolt://localhost:7687";
 const NEO4J_USER = "neo4j";
 const NEO4J_PASSWORD = "password";
 
 const AMQP_URI = "amqp://localhost";
 
-const plugin = new Neo4jGraphQLSubscriptionsAMQPPlugin({
+const plugin = new Neo4jGraphQLSubscriptionsEngineAMQP({
```
Comment (angrykoala) [inline examples/subscriptions/apollo_rabbitmq/server.js:38]: plugin :eyes:

**Review by angrykoala — APPROVED** [review]

**Pull request merged**